src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -116,7 +119,11 @@ public void execute(InputFileDetails inputFileDetails) {
         inputs.removed(new Action<InputFileDetails>() {
             @Override
             public void execute(InputFileDetails inputFileDetails) {
-                getProject().delete(outputFileFor(inputFileDetails.getFile()));
+                File input = inputFileDetails.getFile();
+                if (input.isDirectory()) {
```

Comment:
We could delete removed directories, right? So we don't have empty directories lying around in the output directory.

I am not feeling strongly about this and I guess removing empty directories would also require extra coverage.

Reply:
The plot thickens. The plugin doesn't actually replicate the input directory tree in the output directory so there are no directories to remove.